psutil/arch/freebsd/specific.c

```diff
@@ -31,6 +31,7 @@
 #define PSUTIL_TV2DOUBLE(t)    ((t).tv_sec + (t).tv_usec / 1000000.0)
 #define PSUTIL_BT2MSEC(bt) (bt.sec * 1000 + (((uint64_t) 1000000000 * (uint32_t) \
         (bt.frac >> 32) ) >> 32 ) / 1000000)
+#define SYSCTLTEMP(t) ((int)((t + 270)) - 3000) / 10
```

Comment:
What's the logic here? Where did you take this? Are we returning Celsius? If there's some source code of reference it would be worth mentioning it here as a comment.

Reply:
Yeah.. this bit kinda looks like a hack.. Explanation: 

Calling `sysctlbyname` returns an integer. Usually something like this: `3121`. 

Always ending with `1`, and starts with `3` for the temperatures I have observed. 

By comparing this with `sysctl` calls, I was able to determine that the actual temperature is the middle 2 digits (in this case `12`) with and offset of `27`, so in this case: `39`. 

I wasn't able go find official documentation of this, so if you or a FreeBSD expert have a better idea of how to get actual values, that would be great. 

Link to `coretemp` implementation: 
https://github.com/freebsd/freebsd/blob/master/sys/dev/coretemp/coretemp.c

Reply:
I see. It looks like coretemp.c implements the conversion logic like this: 
https://github.com/freebsd/freebsd/blob/b3f6f192fe65a06ad95d7d6eef0c58a789e7326f/sys/dev/coretemp/coretemp.c#L372-L375
Ideally I think it would be better to do the same but practically I'm not sure if the same logic can be applied to your code.
I'm CC-ing @glebius, @sunpoet, @kostikbel just in case they know more about this.

Reply:
https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-software-developer-vol-3b-part-2-manual.pdf

Paragraph 14.8 there. Let me read myself :) and come with review .

Reply:
Thanks @glebius.
Here's how netdata implements the conversion logic:
https://github.com/netdata/netdata/blob/545d07eb78222a80124e9187e88921013ee403ea/collectors/freebsd.plugin/freebsd_sysctl.c#L520

Reply:
Okay, I reviewed and I don't understand why do you want to do conversion in the python module? The sysctl already returns Celsius. Looks like on your box it returns some strange data. What version of FreeBSD do you use @amanusk and what is the hardware?

Reply:
@glebius, Thanks for you help with this.
`freebsd-version -k` returns `11.2-RELEASE`

System is Thinkpad T430, with i7-3740QM 

Running `sysctl dev.cpu.0.temperature` in Shell returns: `dev.cpu.0.temperature: 42.0C`  which is indeed in Celsius. The bit where I need to convert is line 1030 in the discussed file (sorry for not knowing how to link line). It returns a 4 digit integer as explained in this conversion.